### PR TITLE
Fix #25917: Template Builder file containers were saving with Identifier instead of Path

### LINKED ISSUE #25917: Template Builder: File containers are not getting saved correctly
### Problem Statement
Currently when creating file containers in the template builder, the containers do not save correctly.

### Steps to Reproduce
1. Create a template
2. Add a file container
3. Create a page with this template
4. Add content
5. Edit the template and add the same container
6. See how the content in the page gets duplicated

The problem is we are sending the `identifier` for the new file containers and the endpoint expects the `path`.

### Acceptance Criteria
Content in a page should not be duplicated

### External Links 
In the old template builder, this is how we update the id:

https://github.com/dotCMS/core/blob/master/core-web/apps/dotcms-ui/src/app/api/services/dot-template-containers-cache/dot-template-containers-cache.service.ts#L29

### Assumptions & Initiation Needs
- The issue is isolated to the file containers not saving correctly.
- The solution requires the path to be sent as the id.

### Quality Assurance Notes & Workarounds
- Test that the file containers are now saving correctly.
- If there are any issues, check to make sure the path is being sent as the id.

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
📁🛠️🧪

The pull request adds support for file containers in the template builder. File containers are containers defined by a file path instead of an identifier. The pull request modifies several files to handle file containers consistently and adds tests and mock data for the new functionality.

> _To build templates with file containers_
> _We need some new code maintainers_
> _They added `DotContainer`_
> _And `getContainerReference`_
> _And mocked some data for the trainers_

### Walkthrough
*  Updated the import path for `DotPipesModule` to use the correct alias ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-3bfbaa3dfc6bc3b5a0c2508554db0ca7e6b90a96248b5993b6a4ab7b9b5b5a1aL5-R6))
*  Added the `DotMessagePipe` as a provider for the `DotPaletteContentTypeModule` to translate content type labels ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-3bfbaa3dfc6bc3b5a0c2508554db0ca7e6b90a96248b5993b6a4ab7b9b5b5a1aR13))
*  Imported the `getContainerReference` function from `gridstack-utils.ts` to the template builder store ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abL25-R26))
*  Used the `getContainerReference` function to handle file containers in the store state ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abR354-R355), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abL362-R370), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abR411-R412), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abL418-R423), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-a32a0fcbcf1b026389eade1b0cd88e8838bbfacb92905fe9da2bea7f72b7c9abL431-R436))
*  Added the `getContainerReference` function to the `gridstack-utils.ts` file to return the identifier or path of a container based on its source ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-f3792a46b295fb618b6d4a8d8f4ecef72ec5e3c41a470d6a44886881a6b93659L3-R3), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-f3792a46b295fb618b6d4a8d8f4ecef72ec5e3c41a470d6a44886881a6b93659R269-R280))
*  Added the `source` and `path` properties to the container mocks in the `dot-containers.mock.ts` file ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-0ecfe3a372abab271a48300d070f7041c306166baadb86c59b1565f19ee53697L155-R164))
*  Modified the `addContainer` function in the `template-builder.store.spec.ts` file to accept a container parameter ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-31b7a396b598816d0a94599edd0c1338d80f276fed9775642769710040776e24L49-R53))
*  Added the `mockFileContainer` variable to the `template-builder.store.spec.ts` file to test file containers ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-31b7a396b598816d0a94599edd0c1338d80f276fed9775642769710040776e24L38-R46))
*  Added test cases for adding file containers to the sidebar and to specific boxes in the `template-builder.store.spec.ts` file ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-31b7a396b598816d0a94599edd0c1338d80f276fed9775642769710040776e24R438-R446), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-31b7a396b598816d0a94599edd0c1338d80f276fed9775642769710040776e24R478-R490))
*  Added a test suite for the `getContainerReference` function in the `gridstack-utils.spec.ts` file ([link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-3647ec392c96563c5712834ab3a47c47021cc2e0b7aa88a8add5bd49849a6a42L3-R7), [link](https://github.com/dotCMS/core/pull/25920/files?diff=unified&w=0#diff-3647ec392c96563c5712834ab3a47c47021cc2e0b7aa88a8add5bd49849a6a42R130-R143))

#### Original
https://github.com/dotCMS/core/assets/63567962/a072893a-24fb-4b83-b8a0-22bb42d922dc  

#### Updated
https://github.com/dotCMS/core/assets/63567962/beb8a09c-b292-4b8f-8326-9039db84904b

### CODE REVIEW

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts [hunk 1]
```diff
@@ -266,3 +266,15 @@ function simulateRowSpace(boxes: DotGridStackWidget[] = [], newBox?: DotGridStac
...
+ * Based on the container source, it returns the identifier that should be used as reference.
+ *
+ * @param dotContainer
+ * @returns string
+ */
+export function getContainerReference(dotContainer: DotContainer): string {
+    return dotContainer.source === CONTAINER_SOURCE.FILE
+        ? dotContainer.path
+        : dotContainer.identifier;
+}
```
Comment (fmontes) [inline core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts:280]: This is not gridstack, this is very dotcms.

File: core-web/libs/template-builder/src/lib/components/template-builder/store/template-builder.store.ts [hunk 2]
```diff
@@ -350,6 +351,8 @@ export class DotTemplateBuilderStore extends ComponentStore<DotTemplateBuilderSt
 
         if (!container) return state;
 
+        const containerReference = getContainerReference(container);
```
Comment (fmontes) [inline core-web/libs/template-builder/src/lib/components/template-builder/store/template-builder.store.ts:354]: If you do this in the container selector we don't have to add it anywhere else.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/template-builder/src/lib/components/template-builder/store/template-builder.store.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/libs/template-builder/src/lib/components/template-builder/store/template-builder.store.ts:354]: Yes! Going to do it that way

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts:280]: moved!

**Review by fmontes — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**